web-console/src/components/supervisor-statistics-table/supervisor-statistics-table.tsx

```diff
@@ -0,0 +1,196 @@
+import { downloadFile, QueryManager } from '../../utils';
+
+import './supervisor-statistics-table.scss';
+
+export interface SupervisorStatisticsTableProps {
+  endpoint: string;
+  transform?: (x: any) => any;
+  downloadFilename?: string;
+}
+
+export interface SupervisorStatisticsTableState {
+  data: JSON[];
+  jsonValue: string;
+  loading: boolean;
+  error: string | null;
+}
+
+export class SupervisorStatisticsTable extends React.PureComponent<
+  SupervisorStatisticsTableProps,
+  SupervisorStatisticsTableState
+> {
+  private supervisorStatisticsTableQueryManager: QueryManager<null, SupervisorStatisticsTableState>;
+  constructor(props: SupervisorStatisticsTableProps, context: any) {
+    super(props, context);
+    this.state = {
+      data: [],
+      jsonValue: '',
+      loading: true,
+      error: null,
+    };
+    this.supervisorStatisticsTableQueryManager = new QueryManager({
+      processQuery: async () => {
+        const { endpoint, transform } = this.props;
+        const resp = await axios.get(endpoint);
+        let data = resp.data;
+        if (transform) data = transform(data);
+        return data;
+      },
+      onStateChange: ({ result, loading, error }) => {
+        const dataArray: JSON[] = [];
+        if (result) {
+          Object.keys(result).forEach(key => {
+            dataArray.push(result[key]);
+          });
+        }
+        this.setState({
+          jsonValue: result ? JSON.stringify(result) : '',
+          data: dataArray,
+          loading,
+          error,
+        });
+      },
+    });
+  }
+
+  componentDidMount(): void {
+    this.supervisorStatisticsTableQueryManager.runQuery(null);
+  }
+
+  renderCell(data: any) {
+    return (
+      <div>
+        <div>{`Processed: ${data.processed.toFixed(1)}`}</div>
+        <div>{`Unparseable: ${data.unparseable.toFixed(1)}`}</div>
+        <div>{`ThrownAway: ${data.thrownAway.toFixed(1)}`}</div>
+        <div>{`ProcessedWithError: ${data.processedWithError.toFixed(1)}`}</div>
+      </div>
+    );
+  }
+
+  renderTable(error: string | null) {
+    const columns = [
+      {
+        Header: 'Task Id',
+        id: 'task_id',
+        accessor: (d: {}) => Object.keys(d)[0],
+      },
+      {
+        Header: 'Totals',
+        id: 'totals',
+        accessor: (d: any) => d[Object.keys(d)[0]].totals.buildSegments,
+        Cell: (d: any) => {
+          return this.renderCell(d.value);
+        },
+      },
+      {
+        Header: '1m',
+        id: '1m',
+        accessor: (d: any) => d[Object.keys(d)[0]].movingAverages.buildSegments['1m'],
+        Cell: (d: any) => {
+          return this.renderCell(d.value);
+        },
+      },
+      {
+        Header: '5m',
+        id: '5m',
+        accessor: (d: any) => d[Object.keys(d)[0]].movingAverages.buildSegments['5m'],
+        Cell: (d: any) => {
+          return this.renderCell(d.value);
+        },
+      },
+      {
+        Header: '15m',
```

Comment:
Ideally these columns would be auto generated based on the content of `.movingAverages.buildSegments` instead of hardcoding 5m, 15m etc